### PR TITLE
Fee Handler

### LINKED ISSUE #57: Add Whitelist contract
Subtask of #6 

For building and running tests of transaction with fee currencies, we need the whitelist contract to query valid fee currencies.

```[tasklist]
### Tasks
- [ ] Add the whitelist contract to the https://github.com/celo-org/optimism repo
- [ ] Adapt the import script and generate wrappers for it: https://github.com/celo-org/op-geth/tree/celo1/contracts/celo
```

### PR DESCRIPTION
**Pull request by pahor167**
Introduces:

FeeHandler
FeeCurrencyWhitelist
MentoFeeHandlerSeller
UniswapFeeHandlerSeller
SortedOracles
AddressSortedLinkedListWithMedian

to devnet

Closes https://github.com/celo-org/optimism/issues/57

### CODE REVIEW

File: op-bindings/predeploys/addresses.go [hunk 1]
```diff
@@ -25,8 +25,15 @@ const (
...
-	GoldToken                     = "0x471ece3750da237f93b8e339c536989b8978a438"
+
+	CeloRegistry                      = "0x000000000000000000000000000000000000ce10"
+	GoldToken                         = "0x471ece3750da237f93b8e339c536989b8978a438"
+	FeeHandler                        = "0xcd437749e43a154c07f3553504c68fbfd56b8778"
+	FeeCurrencyWhitelist              = "0xbb024e9cdcb2f9e34d893630d19611b8a5381b3c"
+	MentoFeeHandlerSeller             = "0x4efa274b7e33476c961065000d58ee09f7921a74"
+	UniswapFeeHandlerSeller           = "0xd3aee28548dbb65df03981f0dc0713bfcbd10a97"
+	SortedOracles                     = "0xefb84935239dacdecf7c5ba76d8de40b077b7b33"
+	AddressSortedLinkedListWithMedian = "ED477A99035d0c1e11369F1D7A4e587893cc002B"
```
Comment (karlb) [inline op-bindings/predeploys/addresses.go]: With the exception of the Registry and GoldToken, I don't think there is a need to preserve the current addresses, as those contracts usually (always?) used internally. But keeping the addresses also won't hurt (except for making them less easy to recognize compared to addresses like `0x4200000000000000000000000000000000000021`).

Comment (pahor167) [reply op-bindings/predeploys/addresses.go]: Personally I would prefer to keep it like this at least for now since it is easier to troubleshoot any problems with devnet (it can be compared to working mainnet). Since we should be registry to get smart contract addresses, it shouldn't be any problem to change it in later stages.

File: op-bindings/foundry/artifact.go
Comment (karlb) [inline op-bindings/foundry/artifact.go:1]: Why do we need these changes? The actual linking is done by forge, so why can't we keep `LinkReferences` as a `RawMessage`?

And if it is really necessary to make linking work, should we upstream it to optimism?

Comment (pahor167) [reply op-bindings/foundry/artifact.go:1]: Thanks for noticing, this is a leftover from one of the trials - reverted

File: op-bindings/gen/main.go [hunk 2]
```diff
@@ -136,10 +136,11 @@ func main() {
 		if err := os.WriteFile(abiFile, rawAbi, 0o600); err != nil {
 			log.Fatalf("error writing file: %v\n", err)
 		}
-		rawBytecode := artifact.Bytecode.Object.String()
```
Comment (karlb) [inline op-bindings/gen/main.go]: Why is this change needed?

Comment (pahor167) [reply op-bindings/gen/main.go]: removed

**Review by karlb — APPROVED** [review]
Looks good to me. I didn't check the contracts code, assuming they are not meaningfully changed.
